SWLOR.Game.Server/Feature/AIDefinition/GenericAIDefinition.cs

```diff
@@ -32,7 +32,106 @@ static float CalculateAverageHP(uint creature)
             var lowestHPAllyRace = GetRacialType(lowestHPAlly);
             var allyCount = allies.Count;
             var activeConcentration = Ability.GetActiveConcentration(self).Feat;
-            
+
+            // Benevolence
+            if (CheckIfCanUseFeat(self, target, FeatType.Benevolence1, () => allyHPPercentage < 100))
+            {
+                return (FeatType.Benevolence1, lowestHPAlly);
+            }
+            if (CheckIfCanUseFeat(self, target, FeatType.Benevolence2, () => allyHPPercentage < 100))
```

Comment:
Benevolence 1 is going to take priority over 2 and 3. I don't think 2 or 3 will run at all if 1 is ready. This might be a place where you want to check the percentages. If they're down to 80% or higher, use 1. If they're between 50-79%, use 2, etc. Just a thought - maybe this was intentional for a reason that isn't clear to me.

Reply:
I only give NPC's one version of these abilities, and they have sufficient FP to not worry about resource management, so it's not an issue. I could reverse them and set up different priorities but with how long it takes certain skills to actually get used I don't see a ton of value in this as by the time an NPC starts to heal after taking damage it's probably at 70% or less.

Reply:
I'm thinking more long term here. The same AI will be used for player companions (animals, droids, whatever) so from that perspective it'd be good to reorganize this if you don't mind. It'll also protect us if someone doesn't know about this and happens to add multiple levels. Less confusion across the board if we just get it right here in one spot.

Reply:
True, I'll get all these swapped around so they're in reverse order.